clang/lib/Basic/Targets/SPIR.h

```diff
@@ -41,12 +41,12 @@ static const unsigned SPIRAddrSpaceMap[] = {
 };
 
 static const unsigned SYCLAddrSpaceMap[] = {
-    4, // Default
+    0, // Default
     1, // opencl_global
     3, // opencl_local
     2, // opencl_constant
     0, // opencl_private
-    4, // opencl_generic
+    0, // opencl_generic
```

Comment:
I have some doubts about having both 0 for `opencl_private` and `opencl_generic`.

Why does it works at the end? Actually does it work on real hardware? At least we should check that it fits what is consumed by Vitis through our massaging passes.

Reply:
I will need to do some more testing on this.
but this prevents a errors during compilation.